.gitlab-ci.yml

```diff
@@ -428,19 +428,41 @@ check-polkadot-companion-status:
   script:
     - ./.maintain/gitlab/check_polkadot_companion_status.sh
 
-check-polkadot-companion-build:
-  stage:                           build
+.check-companion-build:            &check-companion-build
+  stage:                           test
   <<:                              *docker-env
   <<:                              *test-refs-no-trigger
-  needs:
-    - job:                         test-linux-stable-int
-      artifacts:                   false
   script:
-    - ./.maintain/gitlab/check_polkadot_companion_build.sh
+    - ./.maintain/gitlab/check_companion_build.sh "$COMPANION_ORG" "$COMPANION_REPO" "$COMPANION_BUILD" "$COMPANION_DEPENDENCY"
   after_script:
-    - cd polkadot && git rev-parse --abbrev-ref HEAD
+    - cd "$COMPANION_REPO" && git rev-parse --abbrev-ref HEAD
   allow_failure:                   true
 
+check-polkadot-companion-build:
+  <<: *check-companion-build
+  variables:
+    COMPANION_ORG: paritytech
+    COMPANION_REPO: polkadot
+    COMPANION_DEPENDENCY: paritytech/grandpa-bridge-gadget
+  allow_failure: true
+
+check-cumulus-companion-build:
+  <<: *check-companion-build
+  variables:
+    COMPANION_ORG: paritytech
+    COMPANION_REPO: cumulus
+    COMPANION_BUILD: "cargo check"
```

Comment:
```suggestion
    COMPANION_BUILD: "cargo check --benches --all --tests"
```

Reply:
How about `cargo check --all-targets --workspace`? 

 `--all` is deprecated in favour of `--workspace` https://doc.rust-lang.org/cargo/commands/cargo-check.html 
`--all-targets` = `--lib --bins --tests --benches --examples`